src/temporal_terms.jl

```diff
@@ -0,0 +1,56 @@
+# Basic Lag:
+
+# syntax:
+"""
+    lag(term, nsteps)
+This `@formula` term is used to introduce lagged variables.
+For example `lag(x,1)` effectively adds a new column containing
+the value of the `x` column from the previous row.
+If there are no such row (e.g. because this is the first row), then the lagged column will
+contain `missing`.
+
+Note: this is only a basic row-wise lag operation.
+It is up to the user to ensure that that there data is sorted by the temporal variable,
+and that their observations are spaced with regular time-steps.
+(Which may require adding extra-rows filled with `missing` values.)
+"""
+lag(term, nsteps) = error("`lag`  should only be used within a @formula");
+
+# type of model where syntax applies:
+const LAG_CONTEXT = Any
+
+# struct for behavior
+struct LagTerm{T<:AbstractTerm} <: AbstractTerm
+    term::T
+    nsteps::Int
+end
+
+Base.show(io::IO, lag::LagTerm) = print(io, "lag($(lag.term), $(lag.nsteps))")
```

Comment:
is this the right way to do this?
Does it it render right for things like categorical variables?
Should I be doing something with mimetypes?

Reply:
That should be OK AFAICT.